libraries/AP_Beacon/AP_Beacon.cpp

```diff
@@ -225,6 +228,136 @@ uint32_t AP_Beacon::beacon_last_update_ms(uint8_t beacon_instance) const
+        uint8_t next_idx;
+        float next_angle;
+        if (get_next_boundary_point(beacon_points, num_beacons, curr_beacon_idx, start_angle, next_idx, next_angle)) {
+            // add boundary point to boundary_sorted array
+            curr_boundary_idx++;
+            boundary_points[curr_boundary_idx] = beacon_points[next_idx];
+            curr_beacon_idx = next_idx;
+            start_angle = next_angle;
+
+            // check if we have a complete boundary by looking for duplicate points within the boundary_sorted
+            uint8_t dup_idx = 0;
+            bool dup_found = false;
+            while (dup_idx < curr_boundary_idx && !dup_found) {
+                dup_found = (boundary_points[dup_idx] == boundary_points[curr_boundary_idx]);
+                if (!dup_found) {
+                    dup_idx++;
+                }
+            }
+            // if duplicate is found, remove all boundary points before the duplicate because they are inner points
+            if (dup_found) {
+                uint8_t num_pts = curr_boundary_idx - dup_idx + 1;
+                if (num_pts > AP_BEACON_MINIMUM_FENCE_BEACONS) {
+                    // success, copy boundary points to boundary array and convert meters to cm
+                    for (uint8_t j = 0; j < num_pts; j++) {
+                        boundary[j] = boundary_points[j+dup_idx] * 100.0f;
+                    }
+                    boundary_num_points = num_pts;
+                    boundary_success = true;
+                } else {
+                    // boundary has too few points
+                    boundary_failure = true;
+                }
+            }
+        } else {
+            // failed to create boundary - give up!
+            boundary_failure = true;
+        }
+    }
+
+    // clear boundary on failure
+    if (boundary_failure) {
+        boundary_num_points = 0;
+    }
+}
+
+// find next boundary point from an array of boundary points given the current index into that array
+// returns true if a next point can be found
+//   current_index should be an index into the boundary_pts array
+//   start_angle is an angle (in radians), the search will sweep clockwise from this angle
+//   the index of the next point is returned in the next_index argument
+//   the angle to the next point is returned in the next_angle argument
+bool AP_Beacon::get_next_boundary_point(const Vector2f* boundary_pts, uint8_t num_points, uint8_t current_index, float start_angle, uint8_t& next_index, float& next_angle)
+{
+    // sanity check
+    if (boundary_pts == nullptr || current_index >= num_points) {
+        return false;
+    }
+
+    // get current point
+    Vector2f curr_point = boundary_pts[current_index];
+
+    // search through all points for next boundary point in a clockwise direction
+    float lowest_angle = M_PI_2;
+    float lowest_angle_relative = M_PI_2;
+    bool lowest_found = false;
+    uint8_t lowest_index;
+    Vector2f north_vec(1.0f,0);
```

Comment:
It only matters because of the comments in the code.

> an increasing angle is generally clockwise though.

Humm, are you referring specifically to ArduPilot? Because usually in math angles are measured in counter-clockwise direction.